app/services/vot/component_expander.rb

```diff
@@ -0,0 +1,32 @@
+module Vot
+  class ComponentExpander
```

Comment:
I kinda feel like, especially since there's so much red over in `parser.rb`, this could just be a method on the parser rather than a separate class. But I don't feel super strongly about it

Reply:
It definitely could be. It is basically the exact same implementation that was there before but with an array to keep track of which components we have expanded. I think the implementation is just a _little_ complicated so to me it overshadowed the other primary responsibilities of the parser (e.g. validating component uniqueness and existence).